Adding jakarta servlet API 6 compatibility

danigiri opened the pull request:
I am trying to use the proxy with jetty 12 ee10, which uses jakarta 6. This change and version bump uses v6
 (context https://en.wikipedia.org/wiki/Jakarta_EE).

All tests pass and package is generated

```
➜  HTTP-Proxy-Servlet : master ✘ :✹ ᐅ  mvn clean compile test package
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  30.635 s
[INFO] Finished at: 2024-04-19T16:48:47+01:00
[INFO] ------------------------------------------------------------------------
```

danigiri committed:
Adding jakarta servlet API 6

dsmiley reviewed (commented):
If the servlet is completely unchanged by this dependency upgrade, why bump the major version number to 3.0?

danigiri commented:
@dsmiley I had the thought that a key dependency version bump to a major number warranted a major version change as well to signal it clearly, but there is no other reason, so if it's more aligned we could just bump a minor

dsmiley commented:
Let's continue to test that the previous version works as well, since it is a key dependency.  
https://github.com/mitre/HTTP-Proxy-Servlet/blob/master/.github/workflows/maven.yml shows versions are specified for Apache HttpClient; same could be done here and likewise with alterations in pom.xml.

danigiri committed:
ensuring compatibility with jakarta 6 and 5

danigiri commented:
Added 5 and 6 to the compatibility testing matrix, tested both manually locally and tests are happy as expected. Went for a closed list of options compatibility (`[]`) as not expecting jakarta versions to change minor numbers that much if at all.

Also moved back to 2.2 as versioning bump, would that be more coherent?

We're leaving it open in the pom and testing with both (and the rest of combinations on the matrix), so we're basically saying that you can use this component with either of the two as a dependency. AFAIU.

danigiri commented:
Looks like there is a format issue with the yaml, looking into it

danigiri commented:
`Error: The template is not valid. .github/workflows/maven.yml (Line: 31, Col: 28): A sequence was not expected`

Having a deeper read of https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

Looks like `include` expands on the matrix, but those are single values only, moved the multiple values to the matrix values and not the include

danigiri committed:
setting jakarta version list in matrix

dsmiley commented on pom.xml:
```diff
@@ -57,7 +59,7 @@
     <dependency>
       <groupId>jakarta.servlet</groupId>
       <artifactId>jakarta.servlet-api</artifactId>
-      <version>5.0.0</version>
+      <version>6.0.0</version>
```
this version is fixed/constant; it doesn't use the sys prop you declared.  See httpclient for how it's done.

dsmiley reviewed (commented):
Feel free to ping me if I don't follow up!  This PR has been waiting...
